client/state/domains/notices/actions.js

```diff
@@ -0,0 +1,64 @@
+import wpcom from 'calypso/lib/wp';
+import {
+	DOMAINS_NOTICES_REQUEST,
+	DOMAINS_NOTICES_REQUEST_FAILURE,
+	DOMAINS_NOTICES_REQUEST_SUCCESS,
+	DOMAINS_NOTICE_UPDATE,
+	DOMAINS_NOTICE_UPDATE_FAILURE,
+	DOMAINS_NOTICE_UPDATE_SUCCESS,
+} from 'calypso/state/action-types';
+import 'calypso/state/domains/init';
+
+export const requestDomainsNotices = ( domainName ) => ( dispatch ) => {
+	dispatch( {
+		type: DOMAINS_NOTICES_REQUEST,
+		domainName,
+	} );
+
+	return wpcom.req.get( { path: `/me/domains/${ domainName }/notices` } ).then(
+		( { states } ) => {
+			dispatch( {
+				type: DOMAINS_NOTICES_REQUEST_SUCCESS,
+				domainName,
+				notices: states[ domainName ],
```

Comment:
When there's still no state for a domain, `states` is null, so we should use `states?.[ domainName ]` here.

Reply:
Good catch! I didn't notice this because I had manually created the message in advance through api  😅